### PR TITLE
Read the event count from the SCurve data tree

### PR DESCRIPTION
**Pull request by lmoureaux**
Addesses issue #15 by reading the number of events from the S-curve trees.

## Description
The S-curve fit function has the number of events hard-coded to 1000, but can be changed in `ultraScurve`. Detect it from the S-curve tree instead, to enable using data produced with non-default `--nevts`.

The implementation is not as suggested in #15, but will rather read the number of events from the tree. It's IMHO less error-prone: there is no need for an additional parameter to `fitScanData` (and flag to `anaUltraScurve.py`), that could be forgotten. The fit function will "just work" unless supplied with inconsistent trees (for sample merged using `hadd`), in which case an error may even be thrown (depending on the parameters used to generate the `hadd`ed trees).

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

See #15.

## How Has This Been Tested?

The changes were tested using data from http://cmsonline.cern.ch/cms-elog/1005623. Cross-checks:

* The number of failed fits doesn't change (much) with the number of events. The scan at 10000 events fails to fit weird channels with no data on VFAT 9.
* The average `chi^2` of fits doesn't change linearly (as I would expect if the normalization stayed fixed) with the number of events, but is `~sqrt(nevt)` (as I would expect from statistical fluctuations):

| `nevts` | `<chi^2>` | 
|--------:|---------:|
| 10000 | 114 |
| 1000 | 25 |
| 100 | 13 |
| 10 | 10 |

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by jsturdy** [comment]
Is this ready for review (it's a lack of functionality in github that doesn't allow the person creating the PR to request review...)?

**Comment by lmoureaux** [comment]
@jsturdy It's ready for review.

**Comment by bdorney** [comment]
This issue is also pertinent to the conversation: https://github.com/cms-gem-daq-project/vfatqc-python-scripts/issues/113

File: fitting/fitScanData.py [hunk 1]
```diff
@@ -50,7 +55,8 @@ def fit(self):
 
         random = r.TRandom3()
         random.SetSeed(0)
-        fitTF1 = r.TF1('myERF','500*TMath::Erf((TMath::Max([2],x)-[0])/(TMath::Sqrt(2)*[1]))+500',1,253)
+        fitTF1 = r.TF1('myERF','[3]*TMath::Erf((TMath::Max([2],x)-[0])/(TMath::Sqrt(2)*[1]))+[3]',1,253)
+        fitTF1.FixParameter(3, self.Nev / 2.)
```
Comment (bdorney) [inline fitting/fitScanData.py]: I would prefer we did not add to the list of fit parameters of this TF1.  Please instead use:

```
fitTF1 = r.TF1("myERF","%i*TMath::Erf((TMath::Max([2],x)-[0])/(TMath::Sqrt(2)*[1]))+%i"%(self.Nev / 2, self.Nev / 2.),1,253)
```

And remove line 59.

**Comment by lmoureaux** [comment]
Using string manipulation routines to fix parameters, added the fit summaries to the Elog.

**Pull request merged**